core/services/ocr2/plugins/ocr2keeper/evm21/registry.go

```diff
@@ -926,3 +936,82 @@ func (r *EvmRegistry) getBlockHash(blockNumber *big.Int) (common.Hash, error) {
 
 	return block.Hash(), nil
 }
+
+func (r *EvmRegistry) getTxBlock(txHash common.Hash) (*big.Int, common.Hash, error) {
+	txr, err := r.client.TransactionReceipt(r.ctx, txHash)
+	if err != nil {
+		return nil, common.Hash{}, err
+	}
```

Comment:
do we know what client.TransactionReceipt returns when the tx is not found? Does it return an empty txr object or an error?

We need to differentiate here b/w flaky errors and tx not found

Reply:
let's add a todo till then to merge